### ISSUE TITLE
Source dependency resolution should check for changes in a Git repository no more than once per Gradle invocation

### ISSUE THREAD

**Issue by adammurdoch** [gradle/gradle-native#403]
For consistency with binary dependency resolution, and to avoid expensive remote operations, source dependency resolution should perform each remote query no more than once per build. This means:

- List the versions available in each remote repository once per build and cache the results. Currently this is done (at least) once per selector.
- Don't perform a Git update for any working directory. We should check out each revision into its own working directory that is never updated, only garbage collected.

These should also apply when doing multiple resolutions in parallel.

**Comment by big-guy** [comment]
Split the second point to new issue

**Comment by adammurdoch** [comment]
Fixed in https://github.com/gradle/gradle/pull/6116
